packages/@aws-cdk/aws-secretsmanager/lib/secret-rotation.ts

```diff
@@ -210,7 +210,9 @@ export class SecretRotation extends Construct {
       throw new Error('The `masterSecret` must be specified for application using the multi user scheme.');
     }
 
-    const rotationFunctionName = this.node.uniqueId;
+    // Max length of 64 chars, get the last 64 chars
+    const uniqueId = this.node.uniqueId;
+    const rotationFunctionName = uniqueId.substring(Math.max(uniqueId.length - 64, 0), uniqueId.length);
```

Comment:
I wonder if something in `@aws-cdk/core` wouldn't be useful for such cases:
```
this.node.uniqueIdOf(50); Custom length
this.node.uniqueId64; Last 64 chars
```

There are several places in the code where an id derived from unique id is truncated.

Or this in `constructs` now?

Reply:
A more general solution would be nice... and this is blocking me today. :(